omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/converter/openmrs2_2/CodedOrFreeTextConverter.java

```diff
@@ -30,25 +32,37 @@ public class CodedOrFreeTextConverter extends BaseDelegatingConverter<CodedOrFre
 	
 	@Override
 	public SimpleObject asRepresentation(CodedOrFreeText instance, Representation rep) throws ConversionException {
-		SimpleObject codedOfFreeText = new SimpleObject();
-		
-		if (instance.getSpecificName() != null) {
-			ConceptNameResource1_9 conceptNameResource = (ConceptNameResource1_9) Context.getService(RestService.class)
-			        .getResourceBySupportedClass(ConceptName.class);
-			codedOfFreeText.add("specificName", conceptNameResource.asRepresentation(instance.getSpecificName(), rep));
-		}
-		
-		if (instance.getCoded() != null) {
-			ConceptResource2_0 conceptResource = (ConceptResource2_0) Context.getService(RestService.class)
-			        .getResourceBySupportedClass(Concept.class);
-			codedOfFreeText.add("coded", conceptResource.asRepresentation(instance.getCoded(), rep));
+		SimpleObject codedOrFreeText = new SimpleObject();
+		if (rep instanceof CustomRepresentation) {
+			CustomRepresentation customRep = (CustomRepresentation) rep;
+			DelegatingResourceDescription drd = ConversionUtil.getCustomRepresentationDescription(customRep);
+			for (String propertyName : drd.getProperties().keySet()) {
+				DelegatingResourceDescription.Property property = drd.getProperties().get(propertyName);
+				Object o = ConversionUtil.getPropertyWithRepresentation(instance, propertyName, property.getRep());
+				if (o != null) {
```

Comment:
Not sure if this is correct - should I include this even if it is null, since it is specified in the custom rep?

Reply:
Not sure, I would guess skipping in the correct behavior.  At first instinct it might made sense to do something like ' uuid:"" '  but I doubt we'd want to included a nested set of empty properties, so I think just skipping makes sense, unless someone else as an opinion otherwise.